src/hooks/useIsMobile.ts

```diff
@@ -0,0 +1,21 @@
+import { useEffect, useState } from "react";
+
+const getIsMobile = () => window.innerWidth <= 768;
+
+export default function useIsMobile() {
+    const [isMobile, setIsMobile] = useState(getIsMobile());
+
```

Comment:
Introduce new `useIsModal` hook to change the structure of the different components on the following PR